util/curl_engine.cc

```diff
@@ -470,7 +469,7 @@ void CurlEngineImpl::remove_poll(uv_poll_t *poll_handle)
 {
   LOG::trace_in(Utility::curl, "remove_poll()");
 
-  uv_close((uv_handle_t *)poll_handle, [](uv_handle_t *handle) { delete handle; });
```

Comment:
was getting asan new-delete-type-mismatch

Reply:
I'm not sure this code is correct.  In the original code, poll_handle.release() 'took away' the resource from poll_handle's management, but before that, it was essentially an autoptr.  As such, you now need to add a delete operation in the two error blocks after the new, but before line 460, I believe.

Reply:
You are correct, changed back to unique_ptr for now.  Can consider whether poll_handle should be a member variable instead of being allocated in the future.